ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreRecovery.java

```diff
@@ -82,11 +82,14 @@
                           properties.getTimeToLiveInSeconds()));
               messageRecovered.add(info);
             } else {
-              boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(stream);
-              MessageInfo info =
-                  new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
-                      deleteFlag);
-              messageRecovered.add(info);
+              DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(stream);
+              if (deleteRecord.getVersion() == MessageFormatRecord.Delete_Version_V1) {
```

Comment:
not strictly required. I can just construct msginfo of older version for now. When we enabled write of msgInfo in V3, we can make this change. Just for convenience purpose.